cognition_objects/integration.py

```diff
@@ -0,0 +1,254 @@
+        raise HTTPException(status_code=400, detail=f"Invalid state: {state}")
+    integration: CognitionIntegration = CognitionIntegration(
+        id=id,
+        organization_id=org_id,
+        project_id=project_id,
+        created_by=user_id,
+        updated_by=user_id,
+        created_at=created_at,
+        started_at=started_at,
+        finished_at=finished_at,
+        name=name,
+        description=description,
+        tokenizer=tokenizer,
+        state=state,
+        type=integration_type.value,
+        config=integration_config,
+        llm_config=llm_config,
+        delta_criteria={"delta_url": None},
+    )
+    general.add(integration, with_commit)
+
+    return integration
+
+
+def update(
+    id: str,
+    updated_by: Optional[str] = None,
+    name: Optional[str] = None,
+    description: Optional[str] = None,
+    tokenizer: Optional[str] = None,
+    state: Optional[CognitionMarkdownFileState] = None,
+    integration_config: Optional[int] = None,
+    llm_config: Optional[Dict] = None,
+    error_message: Optional[str] = None,
+    started_at: Optional[datetime.datetime] = None,
+    finished_at: Optional[datetime.datetime] = None,
+    last_synced_at: Optional[datetime.datetime] = None,
+    is_synced: Optional[bool] = None,
+    delta_criteria: Optional[Dict[str, str]] = None,
+    with_commit: bool = True,
+) -> CognitionIntegration:
+    integration: CognitionIntegration = get_by_id(id)
+
+    if updated_by is not None:
+        integration.updated_by = updated_by
+    if name is not None:
+        integration.name = name
+    if description is not None:
+        integration.description = description
+    if tokenizer is not None:
+        integration.tokenizer = tokenizer
+    if state is not None:
+        integration.state = state.value
+    if integration_config is not None:
+        integration.config = integration_config
+        flag_modified(integration, "config")
+    if llm_config is not None:
+        integration.llm_config = llm_config
+        flag_modified(integration, "llm_config")
+    if error_message is not None:
+        integration.error_message = error_message
+    if started_at is not None:
+        integration.started_at = started_at
+    if last_synced_at is not None:
+        integration.last_synced_at = last_synced_at
+    if delta_criteria is not None:
+        integration.delta_criteria = delta_criteria
+        flag_modified(integration, "delta_criteria")
+
+    integration.is_synced = is_synced
+    integration.finished_at = finished_at
+
+    general.add(integration, with_commit)
+    return integration
+
+
+def execution_finished(id: str) -> bool:
+    return bool(
+        session.query(CognitionIntegration)
+        .filter(
+            CognitionIntegration.id == id,
+            CognitionIntegration.state.in_(
+                [
+                    CognitionMarkdownFileState.FINISHED.value,
+                    CognitionMarkdownFileState.FAILED.value,
+                ]
```

Comment:
this creates the array on every call of the function. outside as const set FINISHED_STATES would be preferred